src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/Controller/ControllerGeneratorBase.cs

```diff
@@ -106,6 +108,21 @@ protected string ValidateAndGetOutputPath(CommandLineGeneratorModel generatorMod
             return ValidateAndGetOutputPath(generatorModel, generatorModel.ControllerName + Constants.CodeFileExtension);
         }
 
+        private bool IsInArea(string appBasePath, string outputPath)
+        {
+            return outputPath.StartsWith(Path.Combine(appBasePath, "Areas"));
```

Comment:
>Areas [](start = 68, length = 5)

Should be "Areas\\" with trailing backslash, otherwise Areasxxx folder name would also match